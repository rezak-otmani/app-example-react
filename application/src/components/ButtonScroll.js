import React from 'react'
import image1 from '../img/image_link1.png';
import image2 from '../img/image_link2.png';


class ButtonScroll extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
       imgSrc: {image1}
    };
    this.handleMouseOver = this.handleMouseOver.bind(this);
    this.handleMouseOut = this.handleMouseOut.bind(this);
  }

  handleMouseOver() {
    this.setState({
      imgSrc: {image2}
    });
  }

  handleMouseOut() {
    this.setState({
      imgSrc: {image1}
    });
  }

  render() {
    return (
      <div>
        <img onMouseOver={this.handleMouseOver} onMouseOut={this.handleMouseOut} src={this.state.imgSrc}/>
      </div>
    );
  }

}

ButtonScroll.propTypes = {
}

ButtonScroll.defaultProps = {
}

export default ButtonScroll;
