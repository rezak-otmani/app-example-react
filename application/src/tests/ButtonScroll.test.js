import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ButtonScroll from '../components/ButtonScroll';


describe('ButtonScroll Component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ButtonScroll />
    );
  });

        it('it should render ButtonScroll components ', () => {
            expect(wrapper.length).toBe(1);
	
                });

         it('should render all button', () => {
             expect(wrapper.find('button').length).toBe(6);
          });

          it('should have Links to the pages ', () => {

              expect(wrapper.find('Scrollchor').length).toBe(6);
                });

  describe('shows a links for scroll in the page', () => {

          it('shows a link to Home', () => {
             const link = wrapper.find({to: ''});
             expect(link.length).toEqual(1);
           });

            it('shows a link to Product', () => {
             const link = wrapper.find({to: 'product'});
             expect(link.length).toEqual(1);
           });

           it('shows a link to About', () => {
             const link = wrapper.find({to: 'about'});
             expect(link.length).toEqual(1);
           });

             it('shows a link to Featured', () => {
             const link = wrapper.find({to: 'featured'});
             expect(link.length).toEqual(1);
           });

                it('shows a link to Contact', () => {
             const link = wrapper.find({to: 'contact'});
             expect(link.length).toEqual(1);
           });

              it('shows a link to Demo', () => {
             const link = wrapper.find({to: 'demo'});
             expect(link.length).toEqual(1);
           });


});

               it('renders correctly', () => {
                const tree = renderer.create(
                 <ButtonScroll />
                          ).toJSON();
                  expect(tree).toMatchSnapshot();
                });




});
