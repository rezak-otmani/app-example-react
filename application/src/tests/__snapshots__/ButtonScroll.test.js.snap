// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="ui link list"
>
  <a
    className="item"
    href="#"
    onClick={[Function]}
  >
    <button
      className="ui inverted grey mini circular icon button"
      title="Home"
    >
             
    </button>
  </a>
  <br />
  <a
    className="item"
    href="#product"
    onClick={[Function]}
  >
    <button
      className="ui inverted grey mini circular icon button"
      title="Product"
    />
  </a>
  <br />
  <a
    className="item"
    href="#about"
    onClick={[Function]}
  >
    <button
      className="ui inverted grey mini circular icon button"
      title="About"
    />
  </a>
  <br />
  <a
    className="item"
    href="#featured"
    onClick={[Function]}
  >
    <button
      className="ui inverted grey mini circular icon button"
      title="Featured"
    />
  </a>
  <br />
  <a
    className="item"
    href="#contact"
    onClick={[Function]}
  >
    <button
      className="ui inverted grey mini circular icon button"
      title="Contact"
    />
  </a>
  <br />
  <a
    className="item"
    href="#demo"
    onClick={[Function]}
  >
    <button
      className="ui inverted grey mini circular icon button"
      title="Demo"
    />
  </a>
</div>
`;
